### PR TITLE
fixes AIs being able to teleport the id card console inserted cards to their core

### PR DESCRIPTION
**Pull request by silicons**

### CODE REVIEW

**Comment by praisenarsie** [comment]
[s] ?

**Comment by MoreRobustThanYou** [comment]
vroom

**Review by Cyberboss — APPROVED** [review]

File: code/game/machinery/computer/card.dm [hunk 1]
```diff
@@ -537,7 +537,8 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
 /obj/machinery/computer/card/proc/eject_id_scan(mob/user)
 	if(scan)
 		scan.forceMove(drop_location())
-		user.put_in_hands(scan)
+		if(!isAI(user))
```
Comment (ShizCalev) [inline code/game/machinery/computer/card.dm]: `if(!issilicon(user) && Adjacent(user))`

File: code/game/machinery/computer/card.dm [hunk 2]
```diff
@@ -537,7 +537,8 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
 /obj/machinery/computer/card/proc/eject_id_scan(mob/user)
 	if(scan)
 		scan.forceMove(drop_location())
-		user.put_in_hands(scan)
+		if(!isAI(user))
+			user.put_in_hands(scan)
 		playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, 0)
 		scan = null
 	else //switching the ID with the one you're holding
```
Comment (ShizCalev) [inline code/game/machinery/computer/card.dm]: Add this below the else & do the same w/ the else for eject_id_modify
```
if(issilicon(user) || !Adjacent(user))
	return
```

File: code/game/machinery/computer/card.dm [hunk 3]
```diff
@@ -555,7 +556,8 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
 		GLOB.data_core.manifest_modify(modify.registered_name, modify.assignment)
 		modify.update_label()
 		modify.forceMove(drop_location())
-		user.put_in_hands(modify)
+		if(!isAI(user))
```
Comment (ShizCalev) [inline code/game/machinery/computer/card.dm]: `if(!issilicon(user) && Adjacent(user))`

**Review by ShizCalev — CHANGES_REQUESTED** [review]
Fix TK teleporting IDs too please.

**Comment by jughu** [comment]
Feature

**Review by ShizCalev — APPROVED** [review]

**Pull request merged**